src/getExtendedPublicKey.c

```diff
@@ -26,20 +28,31 @@ void ensureParametersAreCorrect(
 	VALIDATE_PARAM(dataLength == dataBuffer[0] * 4 + 1 );
 }
 
-void initializePath(uint8_t *dataBuffer, path_spec_t* pathSpec)
+
+// TODO(ppershing): this is duplicate from deriveAddress!
+static void initializePath(path_spec_t* spec, uint8_t *dataBuffer, size_t dataSize)
 {
-	STATIC_ASSERT((255 - 1 /* length data */ ) / 4 /* sizeof(pathspec.path[i]) */ > MAX_PATH_LENGTH, __bad_length);
+	// Cast length to size_t
+	size_t length = dataBuffer[0];
+	if (length > ARRAY_LEN(spec->path)) {
+		THROW(ERR_INVALID_DATA);
+	}
+	if (length * 4 + 1 != dataSize) {
+		THROW(ERR_INVALID_DATA);
+	}
 
-	pathSpec->length = dataBuffer[0];
+	spec->length = length;
```

Comment:
I really do not like spec as variable name as it is noise name. I would call the pathSpec variable path, but then we would have code like path->path. But I believe both options are much better than spec.